app/src/main/java/io/nekohasekai/sagernet/vpn/DashboardActivity.kt

```diff
@@ -262,59 +257,31 @@ class DashboardActivity : BaseThemeActivity(),
                 transaction.replace(R.id.flFragmentContainer, fragment)
                 transaction.commit()
             }
-            // Reset IVMTN and IVMCI click states
-            ivMtnClicked = false
-            ivMciClicked = false
-            updateIVMtnIcon() // Update the IVMTN icon
-            updateIVMciIcon() // Update the IVMCI icon
+            // Reset PremiumServers click states
+            ivPremiumServersClicked = false
+            updateivPremiumServersIcon() // Update the PremiumServers icon
         }
 
-        // Set an OnClickListener for IVMTN
-        ivMtn.setOnClickListener {
-            ivMtnClicked = !ivMtnClicked // Toggle the IVMTN click state
-            updateIVMtnIcon() // Update the IVMTN icon
+        // Set an OnClickListener for ivPremiumServers
+        ivPremiumServers.setOnClickListener {
+            ivPremiumServersClicked = !ivPremiumServersClicked // Toggle the ivPremiumServers click state
+            updateivPremiumServersIcon() // Update the ivPremiumServers icon
             // Show/hide the MyFragment based on the click state
-            fragmentContainer.visibility = if (ivMtnClicked) View.VISIBLE else View.INVISIBLE
-            if (ivMtnClicked) {
-                AppRepository.filterServersByTag("mtn")
+            fragmentContainer.visibility = if (ivPremiumServersClicked) View.VISIBLE else View.INVISIBLE
+            if (ivPremiumServersClicked) {
+                AppRepository.filterServersByTag("premiumservers")
```

Comment:
`premiumservers` => `premium` is enough

Reply:
Better to write a new method called `getPremiumServer()` and just return all servers that has `is_free=false`

Reply:
do this part together